subgraphs/balancer-v2/src/common/metrics.ts

```diff
@@ -0,0 +1,200 @@
+import { Address, BigDecimal, BigInt, ethereum } from "@graphprotocol/graph-ts";
+import {
+  getOrCreateDex,
+  getOrCreateFinancials,
+  getOrCreateUsageMetricSnapshot,
+  updatePoolDailySnapshot,
+} from "./getters";
+import {BIGDECIMAL_ZERO, SECONDS_PER_DAY} from "./constants";
+import {Account, DailyActiveAccount, _TokenPrice, LiquidityPool, Swap, LiquidityPoolFee} from "../../generated/schema";
+import { calculatePrice, isUSDStable, swapValueInUSD, valueInUSD } from "./pricing";
+import { scaleDown } from "./tokens";
+import { WeightedPool } from "../../generated/Vault/WeightedPool";
+import { getUsdPricePerToken } from "../pricing";
+import {log} from "matchstick-as";
+
+export function updateFinancials(event: ethereum.Event): void {
+  let financialMetrics = getOrCreateFinancials(event);
+  let dex = getOrCreateDex()
+  let totalValueLocked = BIGDECIMAL_ZERO
+  let totalVolumeUsd = BIGDECIMAL_ZERO
+  let totalFeesUsd = BIGDECIMAL_ZERO
+
+  for (let i = 0; i < dex._poolIds.length; i++) {
+    let pool = LiquidityPool.load(dex._poolIds[i])
+    if (pool) {
+      totalValueLocked = totalValueLocked.plus(pool.totalValueLockedUSD)
+      totalVolumeUsd = totalVolumeUsd.plus(pool.totalVolumeUSD)
+      totalFeesUsd = totalFeesUsd.plus(pool._totalSwapFee)
+    }
+  }
+
+  let supplySideRevenue = totalFeesUsd.times(BigDecimal.fromString("0.9"))
+  let protocolSideRevenue = totalFeesUsd.times(BigDecimal.fromString("0.1"))
```

Comment:
can this number be changed via governance or is it hardcoded?

Reply:
nice catch - it is changed via governance